Automattic-Tracks-iOS/Services/EventLoggingUploadQueue.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+open class EventLoggingUploadQueue {
+
+    private let fileManager = FileManager.default
+
+    /// The log file on top of the queue
+    var first: LogFile? {
+
+        guard let url = try? fileManager.contentsOfDirectory(at: storageDirectory).first else {
+            return nil
+        }
+
+        return LogFile(url: url, uuid: url.lastPathComponent)
+    }
+
+    func add(_ log: LogFile) throws {
+        try ensureStorageDirectoryExists()
+        try fileManager.copyItem(at: log.url, to: storageURL(forLog: log))
+    }
+
+    func remove(_ log: LogFile) throws {
+        let url = storageURL(forLog: log)
+        if fileManager.fileExistsAtURL(url) {
+            try fileManager.removeItem(at: storageURL(forLog: log))
+        }
+    }
+
+    var storageDirectory: URL {
+        return fileManager.documentsDirectory.appendingPathComponent("log-upload-queue")
```

Comment:
How do you feel about making the `log-upload-queue` a constructor parameter instead to make this class more customizable? If we do that, we can probably make the a stored property instead as well.

Reply:
This is a good point – rather than allow injecting the title, I think it makes more sense to allow a developer to inject the entire `storageDirectory` – perhaps they'd rather store them in temporary storage rather than in the user documents directory? 

Adjusted in 810b5c9124ad66e6eff3d08c8e714ebf460bf496.

Reply:
That sounds good to me. I am going to resolve this conversation but there are a couple things in the commit I'd like to point out that I'll do separately.